src/Illuminate/Database/Eloquent/Collection.php

```diff
@@ -193,8 +193,12 @@ public function unique($key = null, $strict = false)
      * @param  mixed  $keys
      * @return static
      */
-    public function only($keys)
+    public function only($keys = '*')
     {
+        if (is_null($keys) || $keys === '*') {
+            return $this;
+        }
+
         $dictionary = Arr::only($this->getDictionary(), $keys);
 
         return new static(array_values($dictionary));
```

Comment:
I think it would be just as good with

```php
public function only($keys)
{
    if (is_null($keys)) {
        return $this;
    }
    
    // ...
}
```